modules/ephemeris/src/main/scala/gem/horizons/EphemerisParser.scala

```diff
@@ -47,15 +55,77 @@ object EphemerisParser {
         i <- utc           <~ space
         _ <- solarPresence
         _ <- lunarPresence <~ spaces1
-        c <- coordinates   <~ skipEol
+        c <- coordinates
       } yield (i, c)
 
+    val elementLine: Parser[Ephemeris.Element] =
+      element <~ skipEol
+
     val ephemeris: Parser[Ephemeris] =
       skipPrefix ~> (
-        many(element).map(Ephemeris.fromFoldable[List]) <~ eoe
+        many(elementLine).map(Ephemeris.fromFoldable[List]) <~ eoe
       )
   }
 
+  import impl.{ element, ephemeris, SOE, EOE }
+
+  /** Parses an ephemeris file into an `Ephemeris` object in memory.
+    *
+    * @param s string containing the ephemeris data from horizons
+    *
+    * @return result of parsing the string into an `Ephemeris` object
+    */
   def parse(s: String): ParseResult[Ephemeris] =
-    impl.ephemeris.parseOnly(s)
+    ephemeris.parseOnly(s)
+
+  /** An `fs2.Pipe` that converts a `Stream[F, String]` of ephemeris data from
+    * horizons into a `Stream[F, ParseResult[Ephemeris.Element]]`.
+    *
+    * @tparam F effect to use
+    *
+    * @return pipe for a `Stream[F, String]` into a `Stream[F, ParseResult[Ephemeris.Element]]`
+    */
+  def parsedElements[F[_]]: Pipe[F, String, ParseResult[Ephemeris.Element]] =
+    _.through(fs2.text.lines)
+     .dropThrough(_.trim =!= SOE)
```

Comment:
The problem with buffer size that I reported seems to have something to do with `dropThrough`. If I switch the `dropThrough` for

```
.dropWhile(_.trim =!= SOE)
.drop(1)
```

It works with any buffer size.

Reply:
I think fs2 would appreciate a bug report if you can make a standalone reproduction.